### PR TITLE
Update docs to use batch/v1beta1 cronjobs

### PR DESCRIPTION
**Pull request by soltysh**
@janetkuo I forgot to update docs, so here it goes ptal
@kubernetes/sig-docs-maintainers ptal

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/5475)

### CODE REVIEW

**Comment by zacharysarah** [comment]
LGTM from a docs standpoint. Waiting for tech review from @janetkuo to merge. 👍

File: docs/user-guide/kubectl-conventions.md [hunk 1]
```diff
@@ -38,7 +38,8 @@ In order for `kubectl run` to satisfy infrastructure as code:
 * Deployment, using `extensions/v1beta1` endpoint - use `deployment/v1beta1` (default).
 * Deployment, using `apps/v1beta1` endpoint - use `deployment/apps.v1beta1` (recommended).
 * Job - use `job/v1`.
-* CronJob - use `cronjob/v2alpha1`.
+* CronJob - using `batch/v1beta1` endpoint - use `cronjob/v1beta1`(default).
+* CronJob - using `batch/v2alpha1` endpoint - use `cronjob/v2alpha1` (deprecated).
```
Comment (janetkuo) [inline docs/user-guide/kubectl-conventions.md]: Remove this from docs since it's deprecated (and we don't encourage people to use it)?

**Comment by janetkuo** [comment]
I have one comment and this LGTM otherwise

**Comment by steveperry-53** [comment]
@soltysh Would you like to remove the one line as suggested by @janetkuo?

File: docs/user-guide/kubectl-conventions.md (hunk 1 above)
Comment (soltysh) [reply docs/user-guide/kubectl-conventions.md]: I've always seen this file as an information. We had this for extensions.Jobs with that same information. Especially to give people time to migrate from one to the other, when they do use it. There are not that many places where we can announce deprecation, this is a good place, imho. I'd suggest leaving this as is.

**Comment by soltysh** [comment]
@steveperry-53 I've left a comment, I think it should be left as is.

**Pull request merged**